tomcat/src/main/java/org/apache/coyote/http11/RequestHandler.java

```diff
@@ -0,0 +1,73 @@
+package org.apache.coyote.http11;
+
+import nextstep.jwp.model.LoginValidator;
+import nextstep.jwp.model.SignupManager;
+import org.apache.coyote.http11.session.Session;
+import org.apache.coyote.http11.session.SessionManager;
+
+public class RequestHandler {
+
+    public static final SessionManager SESSION_MANAGER = new SessionManager();
+    public static final String HOME_PAGE = "/index.html";
+    public static final String UNAUTHORIZED_PAGE = "/401.html";
+    public static final String SESSION_ID = "JSESSIONID";
+
+    private final HttpRequest httpRequest;
+
+    private RequestHandler(final HttpRequest httpRequest) {
+        this.httpRequest = httpRequest;
+    }
+
+    public static RequestHandler from(final HttpRequest httpRequest) {
+        return new RequestHandler(httpRequest);
+    }
+
+    public String execute() {
+        if (httpRequest.getHttpMethod() == HttpMethod.GET && (httpRequest.isSameParsedRequestURI("/login")
+                && httpRequest.hasCookie(SESSION_ID))) {
+            return ResponseBody.redirectResponse(HOME_PAGE);
+
+        }
+        if (httpRequest.getHttpMethod() == HttpMethod.POST) {
+            if (httpRequest.isSameParsedRequestURI("/login")) {
+                return branchOfLoginRequest();
+            }
+            if (httpRequest.isSameParsedRequestURI("/register")) {
+                return branchOfRegisterRequest();
+            }
+        }
+
+        return ResponseBody.from(httpRequest.getParsedRequestURI(), HttpStatus.OK).getResponseMessage();
+    }
+
+    private Session findSession() {
+        if (httpRequest.hasCookie(SESSION_ID)) {
+            final String jsessionid = httpRequest.getCookieValue(SESSION_ID);
+            return SESSION_MANAGER.findSession(jsessionid);
+        }
+        final Session session = new Session();
+        SESSION_MANAGER.add(session);
+        return session;
+    }
+
+    private String branchOfRegisterRequest() {
+        final RequestBody requestBody = httpRequest.getRequestBody();
+        SignupManager.singUp(requestBody);
+        return ResponseBody.redirectResponse(HOME_PAGE);
+    }
+
+    private String branchOfLoginRequest() {
+        final Session session = findSession();
+        if (httpRequest.hasCookie(SESSION_ID)) {
+            return ResponseBody.redirectResponse(HOME_PAGE, httpRequest, session.getId());
+        }
+        return login(session.getId());
+    }
+
```

Comment:
쿠키에 JSessionId가 있는 경우와 없는 경우로 크게 나눠보는건 어떨까요?

Reply:
if 로 분기처리한 부분이 쿠키에 JSessionId 가 있는지 없는지 판단하는 로직입니다!
SESSION_ID = JSESSIONID 입니다!

그래서 SESSION_ID 가 있는 경우 로그인 되어 있다고 가정하여 HOME_PAGE로 리다이렉트합니다.
없을 땐 Login 진행합니다!!

Reply:
현재 branchOfLoginRequest메서드의 흐름을 봤을 때,

```suggestion
private String branchOfRequest() {
    if (httpRequest.hasCookie(SESSION_ID)) {
        final String jsessionid = httpRequest.getCookieValue(SESSION_ID);
        final Session session = SESSION_MANAGER.findSession(jsessionid);
        return ResponseBody.redirectResponse(HOME_PAGE, httpRequest, session.getId());
    }
    final Session session = new Session();
    SESSION_MANAGER.add(session);
    return login(session.getId());
}
```
이런 방식으로 간다면 if문으로 쿠키에 JSESSIONID가 있는 것에 대한 검증을 한번만 할 수 있다는 의미였습니다